[WIP] Renode CI

mateusz-holenko opened the pull request:
**DO NOT MERGE IT**

This is a working PR to test Renode tests in CI.

CarlFK committed:
move #endif - fixes ##479

mateusz-holenko committed:
Use custom config & DTS overlay for Zephyr

mithro committed:
Merge pull request #481 from CarlFK/fix479

move #endif - fixes #479

Commit:
platforms: add netv2 platform

Commit:
targets: add netv2 targets

Commit:
make: update pcie header generation

Commit:
firmware: add hdmi2pcie firmware

Commit:
firmware: reboot device using CSR if possible

Commit:
ci: add netv2 test runs

dependabot-preview[bot] committed:
build(deps): bump third_party/liteeth from `4d9e74f` to `f2b3f7e`

Bumps [third_party/liteeth](https://github.com/enjoy-digital/liteeth) from `4d9e74f` to `f2b3f7e`.
- [Release notes](https://github.com/enjoy-digital/liteeth/releases)
- [Commits](https://github.com/enjoy-digital/liteeth/compare/4d9e74f10a3fe7bf71ba9bde50f49689c6458dc5...f2b3f7eeb1a68a4db0d9a3315994374c15e46019)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Commit:
software: add hdmi2pcie V4L2 driver

dependabot-preview[bot] committed:
build(deps): bump third_party/litesata from `db5d2f7` to `b400655`

Bumps [third_party/litesata](https://github.com/enjoy-digital/litesata) from `db5d2f7` to `b400655`.
- [Release notes](https://github.com/enjoy-digital/litesata/releases)
- [Commits](https://github.com/enjoy-digital/litesata/compare/db5d2f7881161ce5b9a10a0ab42555f884b9d7c1...b40065591fa4c41b22da1830ae253a1efd8d3d0b)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

mithro committed:
Merge pull request #249 from timvideos/dependabot/submodules/third_party/liteeth-f2b3f7e

build(deps): bump third_party/liteeth from `4d9e74f` to `f2b3f7e`

mithro committed:
Merge pull request #262 from timvideos/dependabot/submodules/third_party/litesata-b400655

build(deps): bump third_party/litesata from `db5d2f7` to `b400655`

niklasnisbeth committed:
Make ice40_hx8k_b_evn target use heap placer

niklasnisbeth committed:
Remove "Xilinx" from mkimage output

mkimage.py was referring to all FPGA bitstreams as "Xilinx FPGA
bitstreams"

niklasnisbeth committed:
Use quotes instead of backticks in enter-env.sh

mithro committed:
Merge pull request #275 from niklasnisbeth/master

Misc fixes

mithro commented on scripts/test-renode.sh:
```diff
@@ -0,0 +1,90 @@
+	vexriscv | picorv32)
+		;;
+	*)
+        echo "Simulating the current configuration is not supported in Renode yet - skipping the test"
```
Indenting here looks a bit weird.

mateusz-holenko replied:
It was caused by mixed tabs & spaces. Fixed now.

mithro commented on scripts/test-renode.sh:
```diff
@@ -0,0 +1,90 @@
+
+cd $OPT_RENODE_LOCATION
+
+# TODO: those files should be shipped together with Renode
```
Can you add why these are needed?

mateusz-holenko replied:
They are needed to run robot tests. The fact they are not in the conda package is a bug we need to fix.

wgetting the missing files is just a workaround to verify that everything else works as expected

ewenmcneill committed:
EDID: v_sync offset/width are 6-bit values

Fix assumption that both h_sync and v_sync offset/width are 10-bit
values, which resulted in the v_sync offset being sent as if it were
0; h_sync offset/width are 10-bit values, but v_sync offset/width
are 6-bit values.

For a summary of the timing section of EDID bit packing see:

https://en.wikipedia.org/wiki/Extended_Display_Identification_Data#Detailed_Timing_Descriptor

mithro committed:
Merge pull request #482 from ewenmcneill/edid

EDID: v_sync offset/width are 6-bit values

mithro committed:
make.py: Fix image name.

dependabot-preview[bot] committed:
build(deps): bump third_party/litepcie from `c455779` to `06ee416`

Bumps [third_party/litepcie](https://github.com/enjoy-digital/litepcie) from `c455779` to `06ee416`.
- [Release notes](https://github.com/enjoy-digital/litepcie/releases)
- [Commits](https://github.com/enjoy-digital/litepcie/compare/c4557792267f83a6e5ee1cccce3e1e2cd1d5b46e...06ee416e9eedf6e44ac13b89d06c1c96d9c3dd39)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump third_party/migen from `94db729` to `0d0e17a`

Bumps [third_party/migen](https://github.com/m-labs/migen) from `94db729` to `0d0e17a`.
- [Release notes](https://github.com/m-labs/migen/releases)
- [Commits](https://github.com/m-labs/migen/compare/94db7295fd4942d0ee27de1148a6cc7be356329d...0d0e17a5e65fc21a2f08a16b55594739c98cf89e)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

mmicko committed:
Update waxwing pin mappings according to official UCF

mmicko committed:
Added net configuration for waxwing

mmicko committed:
Add emulator_ram section

mmicko committed:
Set notes for different configurations of FTDI chip

mithro committed:
Merge pull request #238 from antmicro/netv2

Add NeTV2 support and HDMI2PCIe

mithro committed:
Merge pull request #295 from timvideos/dependabot/submodules/third_party/migen-0d0e17a

build(deps): bump third_party/migen from `94db729` to `0d0e17a`

mithro committed:
Merge pull request #294 from timvideos/dependabot/submodules/third_party/litepcie-06ee416

build(deps): bump third_party/litepcie from `c455779` to `06ee416`

mithro committed:
travis: Cleanup the allow_failures.

mithro committed:
travis: Also let the HDMI2USB targets fail.

They non-deterministically take >50minutes to fail.

dependabot-preview[bot] committed:
build(deps): bump third_party/litedram from `ac4b339` to `a35a1f7`

Bumps [third_party/litedram](https://github.com/enjoy-digital/litedram) from `ac4b339` to `a35a1f7`.
- [Release notes](https://github.com/enjoy-digital/litedram/releases)
- [Commits](https://github.com/enjoy-digital/litedram/compare/ac4b339a6fd90bb012e7fea72f1d5fd1c367af44...a35a1f7790cba537d52ccc8ae73c4e230a6cba8d)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump third_party/liteeth from `f2b3f7e` to `f532a12`

Bumps [third_party/liteeth](https://github.com/enjoy-digital/liteeth) from `f2b3f7e` to `f532a12`.
- [Release notes](https://github.com/enjoy-digital/liteeth/releases)
- [Commits](https://github.com/enjoy-digital/liteeth/compare/f2b3f7eeb1a68a4db0d9a3315994374c15e46019...f532a12b40648e84cef626e9343f428e5e366fb4)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump third_party/litepcie from `06ee416` to `9357036`

Bumps [third_party/litepcie](https://github.com/enjoy-digital/litepcie) from `06ee416` to `9357036`.
- [Release notes](https://github.com/enjoy-digital/litepcie/releases)
- [Commits](https://github.com/enjoy-digital/litepcie/compare/06ee416e9eedf6e44ac13b89d06c1c96d9c3dd39...935703636b20306a66d3253f34a4473f0e8d2844)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump third_party/litesata from `b400655` to `1e3573b`

Bumps [third_party/litesata](https://github.com/enjoy-digital/litesata) from `b400655` to `1e3573b`.
- [Release notes](https://github.com/enjoy-digital/litesata/releases)
- [Commits](https://github.com/enjoy-digital/litesata/compare/b40065591fa4c41b22da1830ae253a1efd8d3d0b...1e3573b07d382eac50ef764fd839009bf90cb8ce)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump third_party/litex-renode from `f082da2` to `70e884e`

Bumps [third_party/litex-renode](https://github.com/litex-hub/litex-renode) from `f082da2` to `70e884e`.
- [Release notes](https://github.com/litex-hub/litex-renode/releases)
- [Commits](https://github.com/litex-hub/litex-renode/compare/f082da2e1093bf7033b8842a502f21f295591f3f...70e884e88c25d997ea5e3dd1e2a8bb08f0d02f90)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump third_party/flash_proxies from `01d8f81` to `3a06aa8`

Bumps [third_party/flash_proxies](https://github.com/jordens/bscan_spi_bitstreams) from `01d8f81` to `3a06aa8`.
- [Release notes](https://github.com/jordens/bscan_spi_bitstreams/releases)
- [Commits](https://github.com/jordens/bscan_spi_bitstreams/compare/01d8f819f15baf9a8cc5d96945a51e4d267ff564...3a06aa84b62ad24467fb0d2c6ceddf565e9ea447)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump third_party/litescope from `1448b55` to `daf10e9`

Bumps [third_party/litescope](https://github.com/enjoy-digital/litescope) from `1448b55` to `daf10e9`.
- [Release notes](https://github.com/enjoy-digital/litescope/releases)
- [Commits](https://github.com/enjoy-digital/litescope/compare/1448b5581901ba54b764f7a78268892ae8f5efff...daf10e9473fb70b3034e0331ef89005661ac04e0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

mithro committed:
Merge pull request #306 from timvideos/dependabot/submodules/third_party/liteeth-f532a12

build(deps): bump third_party/liteeth from `f2b3f7e` to `f532a12`

dependabot-preview[bot] committed:
Merge pull request #305 from timvideos/dependabot/submodules/third_party/litedram-a35a1f7

dependabot-preview[bot] committed:
Merge pull request #307 from timvideos/dependabot/submodules/third_party/litepcie-9357036

dependabot-preview[bot] committed:
Merge pull request #308 from timvideos/dependabot/submodules/third_party/litex-renode-70e884e

dependabot-preview[bot] committed:
Merge pull request #269 from timvideos/dependabot/submodules/third_party/flash_proxies-3a06aa8

dependabot-preview[bot] committed:
Merge pull request #259 from timvideos/dependabot/submodules/third_party/litescope-daf10e9

dependabot-preview[bot] committed:
Merge pull request #288 from timvideos/dependabot/submodules/third_party/litesata-1e3573b

mithro committed:
Merge pull request #222 from antmicro/zephyr_dts

Add generation of overlay for dts in Zephyr

mithro committed:
Merge pull request #297 from mmicko/mmicko/waxwing_yosys

Linux on Spartan6

mithro committed:
Merge remote-tracking branch 'origin-h2u/master'

dependabot-preview[bot] committed:
build(deps): bump third_party/litepcie from `9357036` to `061418c`

Bumps [third_party/litepcie](https://github.com/enjoy-digital/litepcie) from `9357036` to `061418c`.
- [Release notes](https://github.com/enjoy-digital/litepcie/releases)
- [Commits](https://github.com/enjoy-digital/litepcie/compare/935703636b20306a66d3253f34a4473f0e8d2844...061418c620cb09e32aafc202ed1379a29c49cc17)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump third_party/litedram from `a35a1f7` to `5a90a8b`

Bumps [third_party/litedram](https://github.com/enjoy-digital/litedram) from `a35a1f7` to `5a90a8b`.
- [Release notes](https://github.com/enjoy-digital/litedram/releases)
- [Commits](https://github.com/enjoy-digital/litedram/compare/a35a1f7790cba537d52ccc8ae73c4e230a6cba8d...5a90a8b0dde504d383ffcc3c1ad6e0724caf5713)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
build(deps): bump third_party/migen from `0d0e17a` to `d11565a`

Bumps [third_party/migen](https://github.com/m-labs/migen) from `0d0e17a` to `d11565a`.
- [Release notes](https://github.com/m-labs/migen/releases)
- [Commits](https://github.com/m-labs/migen/compare/0d0e17a5e65fc21a2f08a16b55594739c98cf89e...d11565a8ead28eb5a18d7d4f57abe2a7562cdc8c)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Merge pull request #315 from timvideos/dependabot/submodules/third_party/litepcie-061418c

dependabot-preview[bot] committed:
Merge pull request #316 from timvideos/dependabot/submodules/third_party/litedram-5a90a8b

dependabot-preview[bot] committed:
Merge pull request #317 from timvideos/dependabot/submodules/third_party/migen-d11565a

mithro committed:
Merge pull request #312 from mithro/image-name-fix

make.py: Fix image name.

PiotrZierhoffer committed:
Bump Zephyr SDK version

mithro committed:
Merge pull request #322 from antmicro/zephyr-sdk-bump

Bump Zephyr SDK version

mateusz-holenko committed:
build-zephyr: Generate csr.csv before DTS overlay

mateusz-holenko committed:
ci: Add micropython test

mateusz-holenko committed:
env: Add Renode test dependencies

mateusz-holenko committed:
litex-renode: Bump submodule

mateusz-holenko committed:
ci: Add Renode tests

mithro committed:
Merge pull request #333 from antmicro/fix_building_zephyr

build-zephyr: Generate csr.csv before DTS overlay

dependabot-preview[bot] committed:
build(deps): bump third_party/litedram from `5a90a8b` to `8a46b71`

Bumps [third_party/litedram](https://github.com/enjoy-digital/litedram) from `5a90a8b` to `8a46b71`.
- [Release notes](https://github.com/enjoy-digital/litedram/releases)
- [Commits](https://github.com/enjoy-digital/litedram/compare/5a90a8b0dde504d383ffcc3c1ad6e0724caf5713...8a46b71411656b676611d9bdf441c04ccec3ec21)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Merge pull request #335 from timvideos/dependabot/submodules/third_party/litedram-8a46b71

mateusz-holenko commented:
The current version is working. Closing this one and opening a new one to `litex-buildenv/master`.
